nbconvert/exporters/slides.py

```diff
@@ -90,7 +90,7 @@ def _reveal_url_prefix_default(self):
             warn("Please update RevealHelpPreprocessor.url_prefix to "
                  "SlidesExporter.reveal_url_prefix in config files.")
             return self.config.RevealHelpPreprocessor.url_prefix
-        return 'reveal.js'
+        return 'https://cdnjs.cloudflare.com/ajax/libs/reveal.js/3.1.0'
```

Comment:
Switch this to:


```
return '//cdnjs.cloudflare.com/ajax/libs/reveal.js/3.1.0'
```

to provide a URL that the browser will adhere to HTTPS or HTTP for the user's page.

Reply:
I'm not sure if it will work, because the CDN redirects HTTP requests to HTTPS. Worth a try, though - thanks @rgbkrk

Reply:
I used:

```javascript
{ src: "//cdn.rawgit.com/hakimel/reveal.js/3.5.0/plugin/notes/notes.js",
```

And it seems to work for me... @takluyver can you check that as well?

Reply:
Nope, it still opens the speaker notes view with an https URL, unfortunately.

Additionally, using `//` breaks viewing the slides from a `file://` URL (i.e. opening the file without running a local http server), whereas that works if we specify the protocol.

Reply:
@damianavila the rawgit URL works for now, but I think it will stop working next month, when rawgit starts redirecting all HTTP requests to HTTPS.

I tested with the URL @rgbkrk suggested - the `notes.html` file for speaker view returns a 403, but you can still see that it's opening with an https URL, which is the cause of the 'mixed content' problem.

Reply:
Your only other choice is to put in specific logic to pick which one is best based on how the jupyter server is hosted, looking at `location.origin`. ¯\\\_(ツ)\_/¯

Reply:
Hmmm. Well, thanks for the info!

Reply:
> Nope, it still opens the speaker notes view with an https URL, unfortunately.

I only modified the piece I referenced above (notes.js) keeping in place all the other url references. And it seems to work for me, but...

> @damianavila the rawgit URL works for now, but I think it will stop working next month, when rawgit starts redirecting all HTTP requests to HTTPS.

Yep, it will probably work until next month. Again, can we serve it from the jupyter CDN?

In any case, having to use `jupyter nbconvert my_slide.ipynb -- to slides --reveal-prefix="reveal.js" --post serve` to have working speaker notes does not seems a big change for our users if we document it properly and explain why we change the default option (maybe with a tiny blog post).

Reply:
Right, I forgot - that was what I was asking @rgbkrk and @minrk about: would it be OK to host a copy of reveal.js on cdn.jupyter.org so that we can access it with plain HTTP?

Reply:
I'd prefer to pare down the use of cdn.jupyter.org, especially for things that aren't our own in favor of things like [unpkg](https://unpkg.com/#/). The only reason for cdn.jupyter I can see is the non-npm packages we own (basically just the notebook-classic js/css). But pointing to unpkg or another CDN for reveal.js ought to work.

BTW, isn't it fine to load https resources on an http page, just not the other way around? Plus, https will work for a `file://` page, whereas `//` will not, so I think explicit https is preferable to `//`, unless I'm missing something.

Reply:
The `https://` URL works except for the speaker notes. Speaker notes open an HTML file at a URL based on the one reveal is loaded from (e.g. https://cdn.rawgit.com/hakimel/reveal.js/3.5.0/plugin/notes/notes.html ). That page connects *back* to the server your slides are served from. 

So if the notes page is an HTTP URL, it works. But if the notes page is an HTTPS URL and you're serving the slides with plain HTTP, the browser doesn't like the insecure connection.

But it's a good point that it would also be a problem the other way around - we can't load reveal with HTTP if you serve the slides using HTTPS.

So I think we just have to use HTTPS, and say that you have to use a local copy of reveal.js if you want to use speaker notes.

Reply:
> So I think we just have to use HTTPS, and say that you have to use a local copy of reveal.js if you want to use speaker notes.

👍

Reply:
> So I think we just have to use HTTPS, and say that you have to use a local copy of reveal.js if you want to use speaker notes.

I am OK with that as well after the whole discussion.